include/rdycore.h.in

```diff
@@ -47,20 +47,43 @@ PETSC_EXTERN PetscErrorCode RDyAdvance(RDy);
 
 // Accessing data
 PETSC_EXTERN PetscBool      RDyFinished(RDy);
+
 PETSC_EXTERN PetscErrorCode RDyGetTime(RDy, PetscReal*);
 PETSC_EXTERN PetscErrorCode RDyGetTimeStep(RDy, PetscReal*);
 PETSC_EXTERN PetscErrorCode RDyGetStep(RDy, PetscInt*);
 PETSC_EXTERN PetscErrorCode RDyGetCouplingInterval(RDy, PetscReal*);
-PETSC_EXTERN PetscErrorCode RDySetCouplingInterval(RDy, PetscReal);
+
 PETSC_EXTERN PetscErrorCode RDyGetNumLocalCells(RDy, PetscInt*);
 PETSC_EXTERN PetscErrorCode RDyGetNumBoundaryConditions(RDy, PetscInt*);
-PETSC_EXTERN PetscErrorCode RDyGetNumBoundaryEdges(RDy, PetscInt, PetscInt*);
-PETSC_EXTERN PetscErrorCode RDyGetBoundaryConditionFlowType(RDy, PetscInt, PetscInt*);
-PETSC_EXTERN PetscErrorCode RDySetDirichletBoundaryValues(RDy, PetscInt, PetscInt, PetscInt, PetscReal*);
-PETSC_EXTERN PetscErrorCode RDyGetHeight(RDy, PetscReal*);
-PETSC_EXTERN PetscErrorCode RDyGetXVelocity(RDy, PetscReal*);
-PETSC_EXTERN PetscErrorCode RDyGetYVelocity(RDy, PetscReal*);
-PETSC_EXTERN PetscErrorCode RDySetWaterSource(RDy, PetscReal*);
+PETSC_EXTERN PetscErrorCode RDyGetNumBoundaryEdges(RDy, const PetscInt, PetscInt*);
+PETSC_EXTERN PetscErrorCode RDyGetBoundaryConditionFlowType(RDy, const PetscInt, PetscInt*);
+
+PETSC_EXTERN PetscErrorCode RDyGetHeightOfLocalCell(RDy rdy, const PetscInt size, PetscReal values[size]);
+PETSC_EXTERN PetscErrorCode RDyGetXMomentumOfLocalCell(RDy rdy, const PetscInt size, PetscReal values[size]);
+PETSC_EXTERN PetscErrorCode RDyGetYMomentumOfLocalCell(RDy rdy, const PetscInt size, PetscReal values[size]);
+
+PETSC_EXTERN PetscErrorCode RDyGetXCentroidOfLocalCell(RDy rdy, const PetscInt size, PetscReal values[size]);
+PETSC_EXTERN PetscErrorCode RDyGetYCentroidOfLocalCell(RDy rdy, const PetscInt size, PetscReal values[size]);
+PETSC_EXTERN PetscErrorCode RDyGetZCentroidOfLocalCell(RDy rdy, const PetscInt size, PetscReal values[size]);
+PETSC_EXTERN PetscErrorCode RDyGetManningsNForLocalCell(RDy rdy, const PetscInt size, PetscReal values[size]);
+PETSC_EXTERN PetscErrorCode RDyGetNatIDOfLocalCell(RDy rdy, const PetscInt size, PetscInt values[size]);
+
+PETSC_EXTERN PetscErrorCode RDyGetXCentroidOfBoundaryEdge(RDy rdy, const PetscInt boundary_index, const PetscInt size, PetscReal values[size]);
```

Comment:
The above suggestion can also be applied here, e.g. `RDyGetBoundaryEdgeXs`, `RDyGetBoundaryCellXs`, etc